src/abstractdataframe/io.jl

```diff
@@ -107,10 +107,12 @@ function _show(io::IO, ::MIME"text/html", df::AbstractDataFrame;
     if eltypes
         write(io, "<tr>")
         write(io, "<th></th>")
+        Threads.lock(TYPE2STRING_LOCK)
         for j in 1:mxcol
             s = html_escape(compacttype(eltype(df[!, j])))
             write(io, "<th>$s</th>")
         end
+        Threads.unlock(TYPE2STRING_LOCK)
```

Comment:
Need to use `lock(...) do` or `try... finally` to avoid keeping the lock forever in case of error.